packages/charts/src/state/selectors/is_chart_empty.ts

```diff
@@ -6,11 +6,16 @@
  * Side Public License, v 1.
  */
 
+import { getInternalChartStateSelector } from './get_internal_chart_state';
 import type { GlobalChartState } from '../chart_state';
+import { createCustomCachedSelector } from '../create_selector';
 
 /** @internal */
-export const isInternalChartEmptySelector = (state: GlobalChartState): boolean | undefined => {
-  if (state.internalChartState) {
-    return state.internalChartState.isChartEmpty(state);
-  }
-};
+export const isInternalChartEmptySelector = createCustomCachedSelector(
+  [(globalChartState: GlobalChartState) => globalChartState, getInternalChartStateSelector],
+  (globalChartState, internalChartState): boolean | undefined => {
+    if (internalChartState) {
+      return internalChartState.isChartEmpty(globalChartState);
+    }
+  },
+);
```

Comment:
I see that we are getting the `internalChartState` with `getInternalChartStateSelector` then passing the state again to the internal selectors like `isChartEmpty`.

Is it possible to simplify this such that you could just do something like...

```suggestion
export const isInternalChartEmptySelector = createCustomCachedSelector(
  [getInternalChartStateSelector],
  (internalChartState): boolean | undefined => {
    if (internalChartState) {
      return internalChartState.isChartEmpty();
    }
  },
);
```

Or is the state different between fetching the `internalChartState` and calling its selectors? Or is there caching benefits of doing it this way? Just curios, doesn't need to be changed in this PR.

Reply:
While most of the charts override this as just `() => false`, `isChartEmpty` as an override could also be implemented as a selector by itself, like the heatmap does - that's why it's using the pattern of passing on global state again.

Reply:
Ok great, thanks